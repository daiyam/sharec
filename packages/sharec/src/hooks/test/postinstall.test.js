const { vol } = require('memfs')
const postinstall = require('../postinstall')

describe('hooks > postinstall >', () => {
  beforeEach(() => {
    vol.reset()
  })

  it('should inject postinstall and uninstall hooks to the target project package.json', async () => {
    const packageJson = {
      name: 'awesome-project',
    }
    const dir = {
      '/target/package.json': JSON.stringify(packageJson),
    }
    vol.fromJSON(dir)

    await postinstall('/target')

    expect(
      JSON.parse(vol.readFileSync('/target/package.json', 'utf8')),
    ).toMatchSnapshot()
  })

  it('should not do anything if hooks already injected', async () => {
    const packageJson = {
      name: 'awesome-project',
      scripts: {
        postinstall: 'sharec install',
        preuninstall: 'sharec remove',
      },
    }
    const dir = {
      '/target/package.json': JSON.stringify(packageJson),
    }
    vol.fromJSON(dir)

    await postinstall('/target')

    expect(
      JSON.parse(vol.readFileSync('/target/package.json', 'utf8')),
    ).toMatchSnapshot()
  })
})
