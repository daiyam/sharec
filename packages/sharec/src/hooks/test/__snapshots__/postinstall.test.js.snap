// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks > postinstall > should inject postinstall and uninstall hooks to the target project package.json 1`] = `
Object {
  "name": "awesome-project",
  "scripts": Object {
    "postinstall": "sharec install",
    "preuninstall": "sharec remove",
  },
}
`;

exports[`hooks > postinstall > should not do anything if hooks already injected 1`] = `
Object {
  "name": "awesome-project",
  "scripts": Object {
    "postinstall": "sharec install",
    "preuninstall": "sharec remove",
  },
}
`;
